src/main/java/org/assertj/core/presentation/CompositeRepresentation.java

```diff
@@ -0,0 +1,76 @@
+package org.assertj.core.presentation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.presentation.StandardRepresentation.STANDARD_REPRESENTATION;
+
+public class CompositeRepresentation implements Representation {
+  protected Map<String, Representation> representations;
+
+  /*Initialize the CompositeRepresentation according to a certain correspondence of the "name of representation" and "representation" itself.*/
+  public CompositeRepresentation(Map<String, Representation> representations) {
+    this.representations = representations;
+  }
+
+  /*Initialize the CompositeRepresentation when the user does not have the correspondence of the "name of representation" and "representation" itself yet.*/
+  public CompositeRepresentation() {
+    representations = new HashMap<>();
+  }
+
+  /*Implementation of the `toString` method.*/
+  @Override
+  public String toStringOf(Object object) {
+    for (Map.Entry<String, Representation> entry : representations.entrySet()) {
+      String value = entry.getValue().toStringOf(object);
+      if (value != null)
+        return value;
+    }
+    return STANDARD_REPRESENTATION.toStringOf(object);
```

Comment:
I think we could use a more functional approach:

- in the constructor, in addition to setup the representations map, also setup the representations in a List
- use this list here and stream + map to the get the `toStringOf(object)`
- use findAny
- chain with orElse to return STANDARD_REPRESENTATION.toStringOf(object)

this would only work if we make the class immutable and initialize the representations map in the constructor.